### PR TITLE
Added field name automatic slugification.

### PR DESCRIPTION
**Pull request by n1k0**
This patch adds support for automatic field name generation when a new title is entered for the first time. It also adds unit and functional tests for the `form` reducer and the `EditableField` component.

r=? @almet

### CODE REVIEW

File: formbuilder/components/EditableField.js [hunk 1]
```diff
@@ -13,26 +13,60 @@ function shouldHandleDoubleClick(node) {
...
+function isDefaultFieldName(name) {
+  return /^field_\d{7}$/.test(name);
+}
+
+function slugify(title) {
+  return title
+    .toLowerCase()
+    .trim()
+    .replace(/\s/g, "_")
+    .replace(/[^a-z0-9]/, "");
```
Comment (leplatrem) [inline formbuilder/components/EditableField.js]: https://github.com/jprichardson/string.js/blob/master/lib/string.js#L12-L13 :)

Comment (n1k0) [inline formbuilder/components/EditableField.js]: I'm gonna use https://www.npmjs.com/package/slug

File: formbuilder/reducers/form.js [hunk 2]
```diff
@@ -24,6 +25,10 @@ function clone(obj) {
   return JSON.parse(JSON.stringify(obj));
 }
 
+function unique(array) {
```
Comment (almet) [inline formbuilder/reducers/form.js]: Should we rename this to `deduplicateArray`?

**Comment by almet** [comment]
If I'm not mistaken, the user sees the name of the field *and* its id, and can update both of them.

What about having only the name exposed, and having the id hidden from the user? I believe it's not something the users need to know.

What do you think?

Otherwise, the code changes looks good to me.

**Comment by n1k0** [comment]
> What about having only the name exposed, and having the id hidden from the user? I believe it's not something the users need to know.

Well if you plan on building the right JSON schema for third-party tools requiring proper field naming, that's really important.

I understand for totally non-technical users just wanting a shareable UI it might be unwanted, but I think we shouldn't drive the implementation by their sole needs...

File: formbuilder/reducers/form.js (hunk 2 above)
Comment (n1k0) [inline formbuilder/reducers/form.js]: Huh. I thought `unique` was meaningful enough :)

Comment (almet) [inline formbuilder/reducers/form.js]: unique is not a verb, that's why I was mistaken.

**Comment by almet** [comment]
that works for me :)

**Comment by n1k0** [comment]
I'm landing this. Filed #8 for hiding the name field.

**Pull request merged**